### PR TITLE
allow for optional started channel to receive runc pid

### PR DESCRIPTION
**Pull request by coryb**
We need to be able to send signals to the runc process (not the processes running under runc), and the Monitor currently does not allow for us to learn the pid in order to signal.   This PR adds an optional `Started chan<- int` to the `CreateOpts` and `ExecOpts` to allow for us to receive the pid just after the runc process has started.

The use-case for sending signals directly to the runc process is that we need to send a `SIGWINCH` for resize events when using externally managed pty pair (not using ConsoleSocket).  The current hacky work-around is to use `PidFile`, which will populate the pid of the runc init or exec process which we can then use to parse `/proc/$pid/status` to find the parent-pid (ie the runc process).  

More context: https://github.com/moby/buildkit/pull/1731#discussion_r506074643
cc @tonistiigi

### CODE REVIEW

**Comment by tonistiigi** [comment]
cc @estesp @AkihiroSuda

File: runc_test.go [hunk 1]
```diff
@@ -182,6 +186,78 @@ func TestRuncExecExit(t *testing.T) {
...
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		interrupt(ctx, t, started)
+	}()
+	status, err := sleepRunc.Run(ctx, "fake-id", "fake-bundle", &CreateOpts{
+		Started: started,
+	})
+	if err == nil {
+		t.Fatal("Expected error from Exec, but got nil")
```
Comment (AkihiroSuda) [inline runc_test.go:215]: s/Exec/Run/

Comment (coryb) [reply runc_test.go]: Whoops, thanks.  Fixed.

**Review by AkihiroSuda — APPROVED** [review]

**Review by estesp — APPROVED** [review]
LGTM

**Pull request merged**